crates/cli/src/subcommands/generate/mod.rs

```diff
@@ -357,9 +364,9 @@ impl WasmCtx {
         let (mem, env) = Self::mem_env(&mut caller);
 
         // Read `buffer_len`, i.e., the capacity of `buffer` pointed to by `buffer_ptr`.
-        let buffer_len = u32::read_from(mem, buffer_len_ptr)?;
+        let buffer_len = u32::from_le_bytes(deref_slice(mem, buffer_len_ptr, 4)?.try_into().unwrap());
         // Write `buffer` to `sink`.
-        let buffer = mem.deref_slice(buffer_ptr, buffer_len)?;
+        let buffer = deref_slice(mem, buffer_ptr, buffer_len)?;
         env.sink.extend(buffer);
 
```

Comment:
I think this should belong to a common method like was with `u32::read_from(mem, buffer_len_ptr)?;` so is easier to keep in sync the logic.

Reply:
I agree, but the whole point of this is that there's no longer a common crate where it would make sense to put this logic, and it's not like we're gonna want to change the way we read a little-endian u32 -- the code that's being duplicated is very simple.

Reply:
Ah, wait, I think I see what you mean, sure.